src/lib/CMakeLists.txt

```diff
@@ -581,6 +583,23 @@ if (${ENABLE_NUMA_SUPPORT})
     set(LIBRARIES ${LIBRARIES} ${NUMA_LIBRARY} hpinuma_msource_s)
 endif()
 
+# Generate a macro file containing probe definitions for dtrace users. As soon as a probe is not present
+# in `provider.h` a compile error would be thrown. With this ugly solution we can avoid this.
+# First we find all probes within the code (e.g. DTRACE_PROBE1(HYRISE, TEST, arg1))
+# Than the regex greps provider and probe name and concactenates them with '_' (e.g. HYRISE_TEST)
+# The probes are than written into a file in order to define the probes
+
+set(PROBES_FILE "${CMAKE_SOURCE_DIR}/src/lib/utils/tracing/probe_definitions.hpp")
+set(LINE "#ifndef &\\n#define &(...)\\n#endif\\n")
+add_custom_command (
+    OUTPUT ${PROBES_FILE}
+    DEPENDS ${SOURCES}
+    COMMAND grep -r --include=*.[ch]pp --exclude=probes.hpp --exclude=probe_definitions.hpp -h DTRACE_PROBE ${CMAKE_SOURCE_DIR} |  sed "s/[^(]*(\\([^,]*\\), \\([^(,|))]*\\).*/\\1_\\2/" | uniq | xargs -n 1 -I & printf "${LINE}" > ${PROBES_FILE}
```

Comment:
This also includes probes that are commented out. Also, please add a comment to the regexp.